UnityProject/Assets/Scripts/Construction/Managers/Deconstruction.cs

```diff
@@ -24,11 +24,12 @@ public void TryTileDeconstruct(TileChangeManager tileChangeManager, TileType til
 	public void ProcessDeconstructRequest(GameObject player, GameObject matrixRoot, TileType tileType,
 		Vector3 cellPos, Vector3 worldCellPos)
 	{
-		if (Vector3.Distance(player.transform.position, worldCellPos) > 1.5f)
+		if (PlayerScript.IsInReach(player.transform.position, worldCellPos) == false)
```

Comment:
transform.position/localPosition aren’t good references for player or items/objects. they can represent out of date positions or not work properly across different matrices.
```suggestion
		if (player.Player().Script.IsInReach(worldCellPos, true) == false)
```
This one uses RegisterTile under the hood, so it’s fine to use